src/goto-symex/execution_state.cpp

```diff
@@ -269,6 +269,16 @@ void execution_statet::symex_step(reachability_treet &art)
       end_thread();
       force_cswitch();
     }
+    else if(
+      instruction.function == "c:@F@main" &&
+      !options.get_bool_option("deadlock-check"))
```

Comment:
The change here is part of the case to deal with goto "END_FUNCTION" instruction. 

While looking at this change in conjunction with issue https://github.com/esbmc/esbmc/issues/152, the change here appears to implement the idea --- if we don't specify "deadlock-check" option, ESBMC will kill the threads to avoid exploring further, which hence preventing dereference to the dead local variable in main function. 

A question here: 
What about the case `(instruction.function == "c:@F@main" && options.get_bool_option("deadlock-check")))`? In other words, will it fail in the same way as in issue https://github.com/esbmc/esbmc/issues/152 if we use ESBMC option "deadlock-check" ?

Reply:
> In other words, will it fail in the same way as in issue #152 if we use ESBMC option "deadlock-check" ?

Good catch. I have added some regression tests with the flag `--deadlock-check`.